redbeams/src/test/java/com/sequenceiq/redbeams/service/network/AwsSubnetChooserTest.java

```diff
@@ -0,0 +1,165 @@
+        List<CloudSubnet> subnets = List.of(
+                new CloudSubnet(SUBNET_1, "", AVAILABILITY_ZONE_A, "", true, false, false),
+                new CloudSubnet(SUBNET_2, "", AVAILABILITY_ZONE_B, "", true, false, false),
+                new CloudSubnet(SUBNET_3, "", AVAILABILITY_ZONE_B, "", true, false, false),
+                new CloudSubnet(SUBNET_4, "", AVAILABILITY_ZONE_C, "", true, false, false)
+        );
+
+        List<CloudSubnet> chosenSubnets = underTest.chooseSubnets(subnets);
+
+        assertThat(chosenSubnets, hasSize(3));
+        assertThat(chosenSubnets, hasItem(allOf(hasProperty("availabilityZone", is(AVAILABILITY_ZONE_A)), hasProperty("id", is(SUBNET_1)))));
+        assertThat(chosenSubnets, hasItem(hasProperty("availabilityZone", is(AVAILABILITY_ZONE_B))));
+        assertThat(chosenSubnets, hasItem(allOf(hasProperty("availabilityZone", is(AVAILABILITY_ZONE_C)), hasProperty("id", is(SUBNET_4)))));
+        verify(awsSubnetValidator).validate(subnets, 2);
+    }
+
+    @Test
+    public void testWithSixPrivateSubnetsFromSomeDifferentAz() {
+        List<CloudSubnet> subnets = List.of(
+                new CloudSubnet("subnet1", "", "us-west-2c", "", true, false, false),
+                new CloudSubnet("subnet2", "", "us-west-2c", "", true, false, false),
+                new CloudSubnet("subnet3", "", "us-west-2b", "", true, false, false),
+                new CloudSubnet("subnet4", "", "us-west-2a", "", true, false, false),
+                new CloudSubnet("subnet5", "", "us-west-2a", "", true, false, false),
+                new CloudSubnet("subnet6", "", "us-west-2b", "", true, false, false)
+        );
+
+        List<CloudSubnet> chosenSubnets = underTest.chooseSubnets(subnets);
+
+        assertThat(chosenSubnets, hasSize(3));
+        assertThat(chosenSubnets, hasItem(hasProperty("availabilityZone", is("us-west-2c"))));
+        assertThat(chosenSubnets, hasItem(hasProperty("availabilityZone", is("us-west-2b"))));
+        assertThat(chosenSubnets, hasItem(hasProperty("availabilityZone", is("us-west-2a"))));
+        verify(awsSubnetValidator).validate(subnets, 2);
+    }
+
+    @Test
+    public void testWithSixPublicSubnetsFromSomeDifferentAz() {
+        List<CloudSubnet> subnets = List.of(
+                new CloudSubnet("subnet1", "", "us-west-2c", "", false, true, true),
+                new CloudSubnet("subnet2", "", "us-west-2c", "", false, true, true),
+                new CloudSubnet("subnet3", "", "us-west-2b", "", false, true, true),
+                new CloudSubnet("subnet4", "", "us-west-2a", "", false, true, true),
+                new CloudSubnet("subnet5", "", "us-west-2a", "", false, true, true),
+                new CloudSubnet("subnet6", "", "us-west-2b", "", false, true, true)
+        );
+
+        List<CloudSubnet> chosenSubnets = underTest.chooseSubnets(subnets);
+
+        assertThat(chosenSubnets, hasSize(2));
+        verify(awsSubnetValidator).validate(subnets, 2);
```

Comment:
Can you please validate that we choose 2 subnet in different az ?

Reply:
fixed